class.jetpack.php

```diff
@@ -2575,6 +2575,35 @@ public static function plugin_deactivation( ) {
 		}
 	}
 
+	/*
+	 * This will clear all the options that the Jetpack connection needs to survive.
+	 *
+	 * This should only be used if:
+	 * 1. wpcom thinks the site is disconnected, AND there are left-over connection options in the local DB.
+	 * (i.e. connection errors like invalid client_id)
+	 *
+	 * 2. An IDC was detected (the site's url has changed), and wpcom was not updated with the new URL changes.
+	 * You would then use this before connecting to spin up a NEW shadow site using this new URL (rather than
+	 * move the connection from the old URL).
+	 *
+	 * @return bool True if success, False if not.
```

Comment:
Are we sure this always returns true? the doc block for `Jetpack_Options::delete_option` says it can return an array of option names.